src/attributes.jl

```diff
@@ -356,6 +356,15 @@ given.
 """
 struct Silent <: AbstractOptimizerAttribute end
 
+"""
+    RawParameter(name::String)
+
+An optimizer attribute for the solver-specific parameter identified by `name`.
+"""
+struct RawParameter <: AbstractOptimizerAttribute
+    name::String
```

Comment:
I'm wondering if `String` is sufficient or if some solvers might want to define enums for their parameters.

Reply:
Would you suggest `RawParameter{T<:Any}` or `RawParameter{T<:Union{AbstractString, Enum}}` ?

Reply:
In JuMP, we would have
```julia
set_parameter(model::Model, param, value) = MOI.set(model, MOI.RawParameter(param), value)
```
so `RawParameter{T<:Any}` would work.

Reply:
Or we could do
```julia
struct RawParameter <: AbstractOptimizerAttribute
    name::Any
end
```
A lot of code that deals with RawParameters won't know what solver its dealing with, so it won't know the value of `T` for collections of RawParameters. We can save the method errors and accept the type instability. It's hard to imagine a case where type instability here causes a performance bottleneck. Additionally, key-value collections of `RawParameter` won't know the type of the values, so there will be type instability in any case.